File: samples/app/src/main/java/com/sourcepoint/example_app/MainActivityAuthId.java
package com.sourcepoint.example_app;

import android.os.Bundle;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import androidx.appcompat.app.AppCompatActivity;

import com.sourcepoint.example_app.core.DataProvider;
import com.sourcepoint.gdpr_cmplibrary.WebViewUtils;

import kotlin.Lazy;

import static org.koin.java.KoinJavaComponent.inject;

public class MainActivityAuthId extends AppCompatActivity {

    private final Lazy<DataProvider> dataProvider = inject(DataProvider.class);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_auth_id);

        WebView wv = findViewById(R.id.webview);
        wv.getSettings().setJavaScriptEnabled(true);

        String authId = dataProvider.getValue().getAuthId();
        String url = dataProvider.getValue().getUrl();

        wv.setWebViewClient(new WebViewClient() {
            @Override
            public void onPageFinished(WebView view, String url) {
                WebViewUtils.setAuthId(authId, view);
            }
        });

        wv.loadUrl(url);
    }
}

File: samples/app/src/main/java/com/sourcepoint/app/v6/MainActivityAuthId.kt
package com.sourcepoint.app.v6

import android.os.Bundle
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.appcompat.app.AppCompatActivity
import com.sourcepoint.app.v6.core.DataProvider
import com.sourcepoint.app.v6.databinding.ActivityMainAuthIdBinding
import com.sourcepoint.cmplibrary.util.setAuthId
import org.koin.android.ext.android.inject

class MainActivityAuthId : AppCompatActivity() {

    private val dataProvider by inject<DataProvider>()
    private lateinit var binding: ActivityMainAuthIdBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainAuthIdBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.webview.apply {
            settings.javaScriptEnabled = true
            val authId: String = dataProvider.authId?:""
            val url: String = dataProvider.url
            webViewClient = object : WebViewClient() {
                override fun onPageFinished(view: WebView, url: String) {
                    setAuthId(authId, view)
                }
            }
            loadUrl(url)
        }
    }
}